### PR TITLE
Added JENDL nuclear data

### PR DESCRIPTION
**Pull request by Shimwell**
Hi all

Perhaps this is not quite ready to merge, I wanted to share this code for downloading and converting JENDL 4.0 nuclear data and ask a question about metastables elements.

The other convert scripts have a section where the script opens the ACE file and changes the ZAID number to ensure it is identified as metastable.

I wanted to know if this is necessary when converting an NJOY file ? It appears that the metastables are identified when the h5 files are produced so I guess not. I just wanted to double check because this is the main difference between this script and the others. 

The other difference is the use of `openmc.data.IncidentNeutron.from_njoy(filename)` which is slower than `openmc.data.IncidentNeutron.from_ace(filename)`

Best,

Jon

### CODE REVIEW

File: convert_jendl.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+import argparse
+import glob
+import os
+import sys
+import tarfile
+import zipfile
+from collections import defaultdict
+from string import digits
+from urllib.request import urlopen
```
Comment (paulromano) [inline convert_jendl.py]: A bunch of these imports are not used: zipfile, defaultdict, digits, urlopen

File: convert_jendl.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                    default='4.0', help="The nuclear data library release version. "
+                    "The only option currently supported is 4.0")
+args = parser.parse_args()
+
+
+
+library_name = 'jendl' #this could be added as an argument to allow different libraries to be downloaded
+endf_files_dir = '-'.join([library_name, args.release, 'endf'])
+# the destination is decided after the release is known to avoid putting the release in a folder with a misleading name
+if args.destination == None:
```
Comment (paulromano) [inline convert_jendl.py]: The Pythonic way to write this is:
```Python
if args.destination is None:
```

File: convert_jendl.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+Are you sure you want to continue? ([y]/n)
+""".format(release_details[args.release]['compressed_file_size'],
+           release_details[args.release]['uncompressed_file_size'])
+
+response = input(download_warning) if not args.batch else 'y'
+if response.lower().startswith('n'):
+    sys.exit()
+
+block_size = 16384
```
Comment (paulromano) [inline convert_jendl.py]: Unused variable

File: convert_jendl.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+for filename in sorted(neutron_files):
+
+    # this is a fix for the TENDL-2017 release where the B10 ACE file which has an error on one of the values
+    if library_name == 'tendl' and args.release == '2017' and os.path.basename(filename) == 'B010':
+        text = open(filename, 'r').read()
+        if text[423:428] == '86843':
+            print('Manual fix for incorrect value in ACE file') # see OpenMC user group issue for more details
+            text = ''.join(text[:423])+'86896'+''.join(text[428:])
+            open(filename, 'w').write(text)
```
Comment (paulromano) [inline convert_jendl.py]: This block here modifying B10 seems unnecessary for JENDL.

**Comment by paulromano** [comment]
One more comment I forgot to put before. I know you fusion guys don't care, but it would be good to include the JENDL thermal scattering sublibrary too. I might make some updates to this script though once merged, so feel free to ignore that and I can take care of it.

File: convert_jendl.py (hunk 1 above)
Comment (Shimwell) [reply convert_jendl.py]: Thanks for this, I noticed a few others in some of the convert scripts so I have also tidied them up

File: convert_jendl.py (hunk 2 above)
Comment (Shimwell) [reply convert_jendl.py]: now fixed, also fixed in convert_tendl and convert fendl

File: convert_jendl.py (hunk 3 above)
Comment (Shimwell) [reply convert_jendl.py]: Now removed, also removed from some other scripts

File: convert_jendl.py (hunk 4 above)
Comment (Shimwell) [reply convert_jendl.py]: Oh this is really bad, sorry for that

**Comment by Shimwell** [comment]
I also spotted a few convert scripts where we are downloading files without using the handy download function in the openmc `_utils.py` script. So I was able to remove a decent amount of lines in these scripts.

**Comment by Shimwell** [comment]
At the moment I am running all the download scripts to check that they work but it takes a while with the njoy parts. I guess this repo is not well suited to CI testing due to the large amounts of data downloading and time required for processing. Perhaps a topic for another PR but in the future we could carry out some automated testing with the different libraries. Perhaps something like this ...

# Run all download scripts within a Circle CI container with OpenMC installed
python3 data/convert_fendl.py
python3 data/convert_jeff32.py
python3 data/convert_nndc71.py
python3 data/convert_tendl.py
python3 data/generate_endf71.py
python3 data/generate_jendl.py

# set OPENMC_CROSS_SECTIONS path to each library 
# run simple sphere model for each isotope in the library with a neutron spectra tally
# flag up inconsistencies with the tally results

**Comment by paulromano** [comment]
Thanks for cleaning up some of the other scripts -- I had been meaning to get rid of the manual downloading!

**Pull request merged**